applications/MPMApplication/custom_conditions/particle_based_conditions/mpm_particle_base_dirichlet_condition.cpp

```diff
@@ -66,6 +66,74 @@ void MPMParticleBaseDirichletCondition::FinalizeSolutionStep( const ProcessInfo&
     m_imposed_displacement = ZeroVector(3);
 }
 
+void MPMParticleBaseDirichletCondition::CalculateInterfaceContactForce(const ProcessInfo& rCurrentProcessInfo )
+{
+    const unsigned int number_of_nodes = GetGeometry().PointsNumber();
+
+    GeneralVariables Variables;
+    MPMShapeFunctionPointValues(Variables.N);
+
+    array_1d<double, 3 > mpc_force = ZeroVector(3);
+
+    for ( unsigned int i = 0; i < number_of_nodes; i++ )
+    {
+        if (Variables.N[i] > std::numeric_limits<double>::epsilon() )
+        {
+            auto r_geometry = GetGeometry();
+            
+            const double nodal_area  = r_geometry[i].FastGetSolutionStepValue(NODAL_AREA, 0);
+            const Vector nodal_force = r_geometry[i].FastGetSolutionStepValue(REACTION);
+
+            if (nodal_area > std::numeric_limits<double>::epsilon())
```

Comment:
```suggestion
            if (nodal_area)
```

##

You may want to throw at least a warning if `NODAL_AREA` vanishes.

Reply:
good suggestion - I added a warning.